### ISSUE TITLE
Print full exception details when DEBUG=True

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-registration#1887]
As a Developer, I want to see the full exception details when an error occurs in an API method, so that I can efficiently debug

the `handle_http_errors()` decorator gobbles all exception thrown at the API level, and prints very minimal information.
We need to check for the Debug flag and print the details so we can have the full details of exceptions when doing local development / in -dev

#### Dependencies

- Blocked by
- Blocking

#### Description:

Brief description of the work involved, including any user research context or key decisions.

#### Acceptance Criteria:

Given
When
Then

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner
